core/capabilities/remote/trigger_subscriber.go

```diff
@@ -106,78 +106,58 @@ func (s *triggerSubscriber) Info(ctx context.Context) (commoncap.CapabilityInfo,
 }
 
 func (s *triggerSubscriber) RegisterTrigger(ctx context.Context, request commoncap.TriggerRegistrationRequest) (<-chan commoncap.TriggerResponse, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
 	rawRequest, err := pb.MarshalTriggerRegistrationRequest(request)
 	if err != nil {
 		return nil, err
 	}
 	if request.Metadata.WorkflowID == "" {
 		return nil, errors.New("empty workflowID")
 	}
```

Comment:
Lock() was here before on purpose - we don't have to unmarshal the request under lock

Reply:
If you can, maybe send out a fix for the race condition in a separate PR, so those two independent changes are not bundled together.